### PR TITLE
Create citizen initiative creation page

### PR DESCRIPTION
**Pull request by illusionOfParadise**

### CODE REVIEW

File: src/Controller/EnMarche/CitizenInitiativeController.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+     */
+    public function createEventAction(Request $request, ?CitizenInitiativeCommand $command): Response
+    {
+        $command = new CitizenInitiativeCommand($this->getUser());
+        $form = $this->createForm(CitizenInitiativeType::class, $command);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $this->get('app.citizen_initiative.handler')->handle($command);
+            $this->addFlash('info', $this->get('translator')->trans('citizen_initiative.creation.success'));
```
Comment (HeahDude) [inline src/Controller/EnMarche/CitizenInitiativeController.php]: Flash messages should be translated in the flash template and they are https://github.com/EnMarche/en-marche.fr/blob/master/templates/base.html.twig#L94.

File: src/Entity/SkillTrait.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace AppBundle\Entity;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+
+trait SkillTrait
+{
+    public function addSkill(Skill $skill): void
```
Comment (HeahDude) [inline src/Entity/SkillTrait.php]: Not sure about that one, this method used to call the reverse side of the ManyToMany.

File: src/Entity/SkillTrait.php [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    /**
+     * @return Skill[]|Collection|iterable
+     */
+    public function getSkills(): iterable
+    {
+        return $this->skills;
+    }
+
+    public function setSkills(?ArrayCollection $skills): void
```
Comment (HeahDude) [inline src/Entity/SkillTrait.php]: No it does not.

File: src/Form/CitizenInitiativeType.php [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+            ->add('coaching_requested', CheckboxType::class, [
+                'required' => false,
+            ])
+            ->add('coaching_request', CoachingRequestType::class, [
+                'required' => false,
+            ])
+        ;
+
+        $builder
+            ->addEventListener(FormEvents::SUBMIT, function (FormEvent $event) {
```
Comment (HeahDude) [inline src/Form/CitizenInitiativeType.php]: Isn't that duplicated code, if it is I+ suggest to encapsulate this logic in a form SkillListener class.

File: src/Form/CitizenInitiativeType.php [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+                $initiative = $event->getForm()->getData();
+                $skills = $event->getData()->getSkills();
+
+                foreach ($skills as $skill) {
+                    if ($skill->getId()) {
+                        continue;
+                    }
+
+                    if ($existingSkill = $this->entityManager->getRepository(Skill::class)->findOneBy(['name' => $skill->getName()])) {
+                        $initiative->replaceSkill($skill, $existingSkill);
```
Comment (HeahDude) [inline src/Form/CitizenInitiativeType.php]: This method should then be part of some SkilledInterface.

File: src/Form/CitizenInitiativeType.php [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+                '15' => '15',
+                '30' => '30',
+                '45' => '45',
+            ],
+        ]);
+    }
+
+    public function getBlockPrefix()
+    {
+        return 'citizen_initiative';
```
Comment (HeahDude) [inline src/Form/CitizenInitiativeType.php]: This is the default, this method does not need to be implemented unless it conflicts with other types.

File: templates/citizen_initiative/add_citizen_initiative.html.twig [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+
+{% block content %}
+<main class="citizen_initiative__create l__wrapper l__row l__row--top l__row--h-stretch">
+    <section class="l__wrapper--main l__wrapper--main-left">
+        <div id="citizen-initiative-form">
+            <p> Vous pouvez désormais créer votre propre initiative citoyenne. Celle sera diffusée aux adhérents autour de vous qui partagent les mêmes centres d’intérêt</p>
+            <h2 class="text--large b__nudge--bottom-small">Je crée mon initiative citoyenne</h2>
+
+            {{ form_start(form, { attr: { class: 'b__nudge--top-40'} }) }}
+            {{ form_errors(form) }}
```
Comment (HeahDude) [inline templates/citizen_initiative/add_citizen_initiative.html.twig]: Wrong indent.

File: src/CitizenInitiative/CitizenInitiativeCommandHandler.php [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+class CitizenInitiativeCommandHandler
+{
+    private $factory;
+    private $manager;
+    private $mailjet;
+
+    public function __construct(
+        EventFactory $factory,
+        ObjectManager $manager,
```
Comment (hhamon) [inline src/CitizenInitiative/CitizenInitiativeCommandHandler.php]: Why?

File: src/CitizenInitiative/Voter/CitizenInitiativeVoter.php [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+        $adherent = $token->getUser();
+        if (null !== $subject || !$adherent instanceof Adherent) {
+            return self::ACCESS_ABSTAIN;
+        }
+
+        if (!in_array(CitizenInitiativePermissions::CREATE, $attributes, true)) {
+            return self::ACCESS_ABSTAIN;
+        }
+
+        return $adherent->isHost() ? self::ACCESS_DENIED : self::ACCESS_GRANTED;
```
Comment (illusionOfParadise) [inline src/CitizenInitiative/Voter/CitizenInitiativeVoter.php]: It includes supervisor

**Pull request merged**